src/main/steps/state-sequence.ts

```diff
@@ -38,52 +38,54 @@ export class StateSequence {
 }
 
 export const currentStateFn = (userCase: Partial<CaseWithId>): StateSequence => {
-  return new StateSequence([
-    State.Draft,
-    State.AwaitingApplicant2Response,
-    State.AwaitingApplicant1Response,
-    State.Applicant2Approved,
-    State.AwaitingHWFDecision,
-    State.AwaitingPayment,
-    State.AwaitingDocuments,
-    State.Submitted,
-    State.AwaitingAos,
-    State.AwaitingService,
-    State.AosDrafted,
-    State.AosOverdue,
-    State.AwaitingServicePayment,
-    State.AwaitingServiceConsideration,
-    State.AwaitingBailiffReferral,
-    State.AwaitingBailiffService,
-    State.IssuedToBailiff,
-    State.GeneralApplicationReceived,
-    State.AwaitingGeneralReferralPayment,
-    State.AwaitingGeneralConsideration,
-    State.GeneralConsiderationComplete,
-    State.AwaitingJudgeClarification,
-    State.AwaitingDwpResponse,
-    State.AwaitingAlternativeService,
-    State.Holding,
-    State.AwaitingConditionalOrder,
-    State.ConditionalOrderDrafted,
-    State.ConditionalOrderPending,
-    State.AwaitingLegalAdvisorReferral,
-    State.AwaitingAdminClarification,
-    State.AwaitingClarification,
-    State.ClarificationSubmitted,
-    State.AwaitingAmendedApplication,
-    State.AwaitingPronouncement,
-    State.ConditionalOrderPronounced,
-    State.ConditionalOrderRefused,
-    State.AwaitingFinalOrder,
-    State.FinalOrderOverdue,
-    State.FinalOrderRequested,
-    State.FinalOrderPending,
-    State.FinalOrderComplete,
-    State.Rejected,
-    State.Withdrawn,
-    State.NewPaperCase,
-    State.OfflineDocumentReceived,
-    State.BulkCaseReject,
-  ]).at(userCase.state as State);
+  return new StateSequence(chronologicalStateSequence).at(userCase.state as State);
```

Comment:
something that could be done as a separate piece of work but StateSequence.at() shouldn't be returning a new StateSequence, as it's being called by the StateSequence itself

Reply:
Created NFDIV-3065.